dataframe/series.go

```diff
@@ -0,0 +1,423 @@
+package dataframe
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"go.starlark.net/starlark"
+)
+
+// Series represents a sequence of values, either ints, floats, or objects. This is
+// the underlying data structure that is used to create DataFrames. A single column
+// of a DataFrame is a Series.
+type Series struct {
+	frozen    bool
+	which     int
+	valInts   []int
+	valFloats []float64
+	valObjs   []string
+	index     []string
+	dtype     string
```

Comment:
I haven't seen cases where `which` & `dtype`  differ, are they always synced? If so It'd be great to make `which` a separate `dtype` go type & give `dtype` a String method instead of carrying around two values. but only if the two can't differ

Reply:
There aren't currently any such cases. However, `dtype` and `which` don't always agree. I don't totally understand how the python version works, and I need to do more research, which is why this may seem a little incomplete. From what I can tell `dtype` can be any arbitrary user-defined string, only some of which have semantic meaning, and is displayed as-is-provided by user code. On the other hand, `which` is being used as an internal only value, which clearly specifies which typed slice contains meaningful values. I can add a TODO explaining as much, but don't think the two should be combined just yet, if ever.

Reply:
perfect. Thanks!